File: 250407/두 수의 차의 최솟값/minimum-difference-between-two-numbers.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int arr[] = new int[10];
        int min_minus = Integer.MAX_VALUE;
        for(int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        for(int i = 0; i < n - 1; i++) {
            int minus = arr[i + 1] - arr[i];
            if(minus < min_minus) {
                min_minus = minus;
            }
        }
        System.out.println(min_minus);
        sc.close();
    }
}

File: 250407/두 수의 차의 최솟값/minimum-difference-between-two-numbers.cpp
#include <iostream>
#include <climits>
using namespace std;

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int n = 0;
    int arr[10] = {0};
    int min_minus = INT_MAX;

    cin >> n;
    for(int i = 0; i < n; i++) {
        cin >> arr[i];
    }
    for(int i = 0; i < n - 1; i++) {
        int minus = arr[i + 1] - arr[i];
        if(minus < min_minus) {
            min_minus = minus;
        }
    }
    cout << min_minus;
    return 0;
}